src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.task.Task;
+
+/**
+ * List of Tasks recorded by the Duke program.
+ * CS2103T iP
+ * AY22/23 Semester 1
+ *
+ * @author Perry Wong
+ */
+public class TaskList {
+
+    private ArrayList<Task> taskArrayList;
+    private int index;
+
+    /**
+     * Constructs a TaskList when there is no previous saved data.
+     */
+    public TaskList() {
+        this.taskArrayList = new ArrayList<>(100);
+        this.index = 0;
+    }
+
+    /**
+     * Constructs a TaskList based on previously saved data.
+     */
+    public TaskList(ArrayList<Task> taskArrayList) {
+        this.taskArrayList = taskArrayList;
+        this.index = taskArrayList.size();
+    }
+
+    /**
+     * Prints the list of tasks that are recorded.
+     */
+    public void printList() {
+        ArrayList<Task> list = this.taskArrayList;
+        list.forEach(t -> System.out.println(list.indexOf(t) + 1 + ". " + list.get(list.indexOf(t)).toString()));
```

Comment:
Although this line is less than 120 char would it be better to still break it into two lines for readability?

Reply:
I'll consider breaking them down into smaller variables as well. Great catch!